src/__experimental__/components/number/number.stories.mdx

```diff
@@ -0,0 +1,208 @@
+  </Story>
+</Preview>
+
+### With labelHelp
+
+<Preview>
+  <Story name="With labelHelp">
+    <Number label="Number" value="123456" labelHelp="Help" />
+  </Story>
+</Preview>
+
+### Required
+
+You can use the `required` prop to indicate if the field is mandatory.
+
+<Preview>
+  <Story name="Required">
+    <Number label="Number" value="123456" required />
+  </Story>
+</Preview>
+
+## Validations
+
+Validation status can be set by passing `error`, `warning` or `info` prop to the component
+
+Passing a string to these props will display a properly colored border along with a validation icon and tooltip - string value will be displayed as the tooltip message.
+
+Passing a boolean to these props will display only a properly colored border.
+
+For more information check our [Validations](?path=/docs/documentation-validations--page "Validations") documentation page
+
+### As a string
+
+<Preview>
+  <Story name="Validations - string - component">
+    {() =>
+      ["error", "warning", "info"].map((validationType) => (
+        <div key={`${validationType}-string-component`}>
+          <Number
+            label="Number"
+            value="123456"
+            {...{ [validationType]: "Message" }}
+            mb={2}
+          />
+          <Number
+            label="Number - readOnly"
+            value="123456"
+            readOnly
+            {...{ [validationType]: "Message" }}
+            mb={2}
+          />
+        </div>
+      ))
+    }
+  </Story>
+</Preview>
+
+### As a string, displayed on label
+
+<Preview>
+  <Story name="Validations - string - label">
+    {() =>
+      ["error", "warning", "info"].map((validationType) => (
+        <div key={`${validationType}-string-label`}>
+          <Number
+            label="Number"
+            value="123456"
+            validationOnLabel
+            {...{ [validationType]: "Message" }}
+            mb={2}
+          />
+          <Number
+            label="Number - readOnly"
+            value="123456"
+            validationOnLabel
+            readOnly
+            {...{ [validationType]: "Message" }}
+            mb={2}
+          />
+        </div>
+      ))
+    }
+  </Story>
+</Preview>
+
+### As a boolean
+
+<Preview>
+  <Story name="Validations - boolean">
+    {() =>
+      ["error", "warning", "info"].map((validationType) => (
+        <div key={`${validationType}-boolean-component`}>
+          <Number
+            label="Number"
+            value="123456"
+            {...{ [validationType]: true }}
+            mb={2}
+          />
+          <Number
+            label="Number - readOnly"
+            value="123456"
+            readOnly
+            {...{ [validationType]: true }}
+            mb={2}
+          />
+        </div>
+      ))
+    }
+  </Story>
+</Preview>
+
+## Props:
+
+### Number
+
+Due to the `Textbox` component being used internally by the `Number` component most of the `Textbox` props can be applied to `Number`
```

Comment:
I think saying "most of" makes it a bit unclear which props can actually be used here - are all the `OriginalTextbox` props valid here?

Reply:
Looking at this again I see we still have the `styleOverride` prop listed in the prop table as well, maybe we should add annotations so it's not exposed as we intend to remove it soon